app/views/RoomView/RightButtons.tsx

```diff
@@ -288,6 +297,31 @@ class RightButtonsContainer extends Component<IRightButtonsProps, IRigthButtonsS
 		showActionSheet({ options });
 	};
 
+	goToNotification = () => {
+		const { room } = this;
+		const { rid, navigation, isMasterDetail } = this.props;
+
+		if (!rid || !room) {
+			return;
+		}
+		if (notificationIsEnabled && room) {
+			if (isMasterDetail) {
+				navigation.navigate('ModalStackNavigator', {
+					screen: 'NotificationPrefView',
+					params: { rid, room }
```

Comment:
I think we talked about this type of approach a time ago.
Here you are passing all the subscription via params to another room, I'm almost sure you don't need all the props, and even if it were necessary the "correct" thing would be to pass only the `asPlain` object.

Reply:
I understood what you commented, however, at the `NotificationPreferencesView` is expected the room with observe. Otherwise, I will need to refactor the `NotificationPreferencesView` too.

Add to this, the `NotificationPrefView` is only enable to open when there is the room provided by the database

Reply:
I replied private

Reply:
https://rocketchat.atlassian.net/browse/TC-991